src/CssVariables.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import propTypes from 'prop-types'
+import * as theme from './theme.js'
+
+const toPrefixedThemeSection = themeSectionKey =>
+    Object.entries(theme[themeSectionKey]).reduce((prefixed, [key, value]) => {
+        prefixed[`${themeSectionKey}-${key}`] = value
+
+        return prefixed
+    }, {})
+
+const toCustomPropertyString = themeSection =>
+    Object.entries(themeSection)
+        .map(([key, value]) => `--${key}: ${value};`)
+        .join('\n')
+
+/**
+ * @module
+ * @description Injects our theme variables as CSS custom properties
+ * @param {CssVariables.PropTypes} props
+ * @returns {React.Component}
+ * @example import { CssVariables } from @dhis2/ui-core
+ * @see Live demo: {@link /demo/?path=/story/cssvariables--default|Storybook}
+ */
+const CssVariables = props => {
+    const variables = Object.keys(props)
+        // Filter all props that are false
+        .filter(prop => props[prop])
+        // Map props to corresponding theme section and prefixes keys with section name
+        .map(toPrefixedThemeSection)
```

Comment:
This will throw an error when providing a prop that doesn't exist on `theme`.
If we want to use this "generic" method, I'd filter out everything that's not in `Object.keys(CssVariables.propTypes)`.

What I like about react is that I can spread object containing the required props on a component while all irrelevant ones are being ignored.

Reply:
Discussed with @ismay, it makes sense to throw an error here to detect typos